<?xml version="1.0" ?>
<launch>

  <arg name="plcf_dir"		default="$(find aist_localization)/plcf"/>
  <arg name="camera_frame"	default="world"/>

  <!-- Spawn depth_filter -->
  <include file="$(find aist_depth_filter)/launch/run.launch">
    <arg name="camera_info_topic"	value="$(arg camera_info_topic)"/>
    <arg name="image_topic"		value="$(arg image_topic)"/>
    <arg name="depth_topic"		value="$(arg depth_topic)"/>
    <arg name="normal_topic"		value="$(arg normal_topic)"/>
    <arg name="manager"			value="$(arg manager)"/>
  </include>

  <!-- Spawn localizer as a node unless manager name is specified -->
  <node if="$(eval manager == '')"
	name="localizer" pkg="aist_localization" type="aist_localization"
  	output="screen">
    <param name="camera_frame"	value="$(arg camera_frame)"/>
    <param name="plcf_dir"	value="$(arg plcf_dir)"/>
  </node>

  <!-- Spawn localizer as a nodelet if manager name is specified -->
  <node unless="$(eval manager == '')"
	name="localizer" pkg="nodelet" type="nodelet" output="screen"
  	args="load aist_localization/aist_localization_nodelet $(arg manager)">
    <remap from="/file_info"	to="depth_filter/file_info"/>
    <param name="camera_frame"	value="$(arg camera_frame)"/>
    <param name="plcf_dir"	value="$(arg plcf_dir)"/>
  </node>

</launch>
